x-pack/plugins/cases/server/client/cases/update.ts

```diff
@@ -497,3 +522,48 @@ const patchCases = async ({
 
   return updatedCases;
 };
+
+const getCasesAndAssigneesToNotifyForAssignment = (
+  updatedCases: SavedObjectsBulkUpdateResponse<CaseAttributes>,
+  casesMap: Map<string, CaseSavedObject>,
+  user: CasesClientArgs['user']
+) => {
+  return updatedCases.saved_objects.reduce((acc, updatedCase) => {
+    const originalCaseSO = casesMap.get(updatedCase.id);
+
+    if (!originalCaseSO) {
+      return acc;
+    }
+
+    const alreadyAssignedToCase = originalCaseSO.attributes.assignees;
+    const comparedAssignees = arraysDifference(
+      alreadyAssignedToCase,
+      updatedCase.attributes.assignees ?? []
+    );
+
+    if (comparedAssignees && comparedAssignees.addedItems.length > 0) {
+      const theCase = mergeOriginalSOWithUpdatedSO(originalCaseSO, updatedCase);
+
+      const assigneesWithoutCurrentUser = comparedAssignees.addedItems.filter(
+        (assignee) => assignee.uid !== user.profile_uid
+      );
+
+      acc.push({ theCase, assignees: assigneesWithoutCurrentUser });
+    }
+
+    return acc;
+  }, [] as Array<{ assignees: CaseAssignees; theCase: CaseSavedObject }>);
```

Comment:
nit: I think we can avoid the cast if we add this definition to the `reduce<Array<{ assignees: CaseAssignees; theCase: CaseSavedObject }>>`